static/js/components/Payment.js

```diff
@@ -19,11 +21,68 @@ export default class Payment extends React.Component {
     payment: RestState,
     payableKlassesData: Array<Object>,
     selectedKlass: Object,
+    now: moment$Moment,
     sendPayment: () => void,
     setPaymentAmount: (event: InputEvent) => void,
     setSelectedKlassKey: (event: InputEvent) => void
   };
 
+  getTotalOwedUpToInstallment = (nextInstallmentIndex: number): number => {
+    const {
+      selectedKlass: { installments }
+    } = this.props;
+
+    return R.compose(
+      R.sum,
+      R.map(R.prop('amount')),
+      R.slice(0, nextInstallmentIndex + 1)
+    )(installments);
+  };
+
+  getDeadlineDates = R.map(
+    R.compose(
+      moment,
+      R.prop('deadline')
+    )
+  );
+
+  hasMissedPreviousInstallment = (previousInstallmentIndex: number): boolean => {
+    const { selectedKlass } = this.props;
+
+    return previousInstallmentIndex >= 0 &&
+      selectedKlass.total_paid < this.getTotalOwedUpToInstallment(previousInstallmentIndex);
+  };
+
+  getPaymentDeadlineText = (): string => {
+    const { selectedKlass, now } = this.props;
+
+    let installments = selectedKlass.installments;
+    let installmentDeadlineText = '',
+      finalInstallmentDeadline = '';
+
+    if (installments.length > 1) {
+      let deadlineDates = this.getDeadlineDates(installments);
```

Comment:
Suggestion: move this line outside the `if` statement, and then remove the `else` block and also move line 77 outside the if statement, since all that code is doing the same thing.